### PR TITLE
Accessibility support targets & tooling setup (#4871)

### LINKED ISSUE #4871: [Discussion] choose appropriate tool for WCAG auto testing / visual regression tests
Problem:
*  Wagtail need a tool for auto frontend tests
* wagtail need auto testing for wcag

Here is the cypress test for wagtail - https://github.com/davojta/test-cypress-for-wagtail

There are several solutions for wcag testing:

* pa11yjs
* axe-core

There also some options for visual regression testing

* cypress
* web driver
* puppeteer 

To be updated later )

### PR DESCRIPTION
**Pull request by thibaudcolas**
This adds a suite of tools and related configuration / documentation, to help us improve Wagtail’s accessibility. Related to #4871, but does not cover the automated integration tests bit. The idea is that once this PR is merged it should be possible for us to make accessibility fixes and meaningfully review UI changes for accessibility issues, even if it's more of a manual process to start with.

Here is what's included:

## Documentation of accessibility support target

I put them alongside the browser support targets for now. These targets are both:

- Ambitious, since we don't formally offer any kind of support at the moment
- Very basic. I chose the assistive technologies that are the easiest to set up and use, and are available at no cost.

I think it's important to have explicit targets so we know where we want to be, nonetheless we can't expect people to actually use all of those tools when making or reviewing changes, so I tried to use the right language to make that clear.

## [jinjalint](https://github.com/motet-a/jinjalint)

Static analysis for Django templates which should help enforcing HTML is well-formed. Its checks aren't comprehensive, but it makes up for that by being much faster than anything aiming for comprehensiveness. (60s to crawl the whole codebase, 200ms per individual template file).

I have decided to discard its output for now (`|true`), because I don't want this tooling PR to be held up by those fixes. I'll do that next though (#5258), so we can start to properly rely on the linter.

For reference, here are the issues it found (malformed HTML, missing end tags, mismatched start/end tags):

```
$ jinjalint --parse-only wagtail/**/*.html
wagtail/admin/templates/wagtailadmin/collection_privacy/ancestor_privacy.html:8:2: Parse error: expected 'p' at 7:2
wagtail/admin/templates/wagtailadmin/edit_handlers/chooser_panel.html:26:150: Parse error: expected '>' at 25:150
wagtail/admin/templates/wagtailadmin/page_privacy/ancestor_privacy.html:8:2: Parse error: expected 'p' at 7:2
wagtail/admin/templates/wagtailadmin/pages/listing/_button_with_dropdown.html:15:6: Parse error: expected 'ul' at 14:6
wagtail/admin/templates/wagtailadmin/pages/listing/_list_explore.html:69:271: Parse error: expected 'p' at 68:271
wagtail/admin/templates/wagtailadmin/permissions/includes/collection_member_permissions_formset.html:43:2: Parse error: expected 'tbody' at 42:2
wagtail/contrib/forms/templates/wagtailforms/list_submissions.html:34:14: Parse error: expected 'td' at 33:14
wagtail/contrib/redirects/templates/wagtailredirects/results.html:19:8: Parse error: expected one of 'autoescape', 'block', 'blocktrans', 'comment', 'filter', 'for', 'if', 'ifchanged', 'ifequal', 'ifnotequal', 'not an intermediate Jinja tag name', 'spaceless', 'verbatim', 'with' at 18:8
wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html:434:205: Parse error: expected one of 'autoescape', 'block', 'blocktrans', 'comment', 'filter', 'for', 'if', 'ifchanged', 'ifequal', 'ifnotequal', 'not an intermediate Jinja tag name', 'spaceless', 'verbatim', 'with' at 433:205
wagtail/documents/templates/wagtaildocs/documents/edit.html:66:5: Parse error: expected one of '[:a-z]', 'area', 'base', 'br', 'col', 'command', 'embed', 'hr', 'img', 'input', 'keygen', 'link', 'meta', 'param', 'script', 'source', 'style', 'track', 'wbr', '{#', '{%', '{{' at 65:5
wagtail/documents/templates/wagtaildocs/multiple/add.html:40:10: Parse error: expected 'p' at 39:10
wagtail/images/templates/wagtailimages/multiple/add.html:41:10: Parse error: expected 'p' at 40:10
wagtail/project_template/home/templates/home/welcome_page.html:6:70: Parse error: expected one of '=', '>', '[0-9-_.]', '[:a-z]', '\\s+', '{#', '{%', '{{' at 5:70
wagtail/snippets/templates/wagtailsnippets/snippets/edit.html:13:11: Parse error: expected one of 'autoescape', 'block', 'blocktrans', 'comment', 'endif', 'filter', 'for', 'if', 'ifchanged', 'ifequal', 'ifnotequal', 'not an intermediate Jinja tag name', 'spaceless', 'verbatim', 'with' at 12:11
wagtail/users/templates/wagtailusers/groups/includes/page_permissions_formset.html:43:2: Parse error: expected 'tbody' at 42:2
wagtail/users/templates/wagtailusers/groups/list.html:18:14: Parse error: expected 'tr' at 17:14
```

## [react-axe](https://github.com/dequelabs/react-axe)

An integration of Axe similar to how it would be used for browser extensions, except that:

- It's part of the build tools, so doesn't require extra setup by individual contributors
- It automatically runs whenever React components render on the page, without having to manually trigger it.
- It de-duplicates issues found between runs, or the same run between components
- Contrary to what the name suggests, it also audits parts of the page that aren't built with React (it just won't do so whenever they dynamically change).

## [Accessibility Insights for Web](https://accessibilityinsights.io/docs/en/web/overview)

Also uses Axe under the hood, and complements its automated checks by very useful semi-manual ones, and a wizard to do manual auditing.

## Axe extension for Chrome

Same as what `react-axe` does, except can be used in production. Also the same as `Accessibility Insights for Web`, but it might be more widely used and is equally as relevant.

----

* [x] Do the tests still pass? (http://docs.wagtail.io/en/latest/contributing/developing.html#testing)
* [x] Does the code comply with the style guide? (Run `make lint` from the Wagtail root)
* ~[ ] For Python changes: Have you added tests to cover the new/fixed behaviour?~
* ~[ ] For front-end changes: Did you test on all of Wagtail’s supported browsers? **Please list the exact versions you tested**.~
* ~[ ] For new features: Has the documentation been updated accordingly?~

### CODE REVIEW

File: docs/contributing/developing.rst [hunk 1]
```diff
@@ -179,6 +185,28 @@ IE             Desktop        10 and below
...
+
+=============  ====================
+Type           Assistive technology
+=============  ====================
+Screen reader  `NVDA <https://www.nvaccess.org/download/>`_ on Windows with Firefox ESR
+Screen reader  `VoiceOver <https://support.apple.com/en-gb/guide/voiceover-guide/welcome/web>`_ on macOS with Safari
+Magnification  `Windows Magnifier <https://support.microsoft.com/en-gb/help/11542/windows-use-magnifier>`_
+Magnification  macOS Zoom
+Voice control  Windows Speech Recognition
+Voice control  macOS Dictation
```
Comment (jonnyscholes) [inline docs/contributing/developing.rst]: Should we mention a mobile screenreader here? My experience is that VoiceOver works near identically on iOS and MacOS so maybe we could include an Android offering? Or is that biting off too much for now...

File: docs/contributing/html_guidelines.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+~~~~~~~~~~~~
+
+We use `jinjalint <https://github.com/motet-a/jinjalint>`_ to lint templates. If you have installed Wagtail's testing dependencies (``pip install -e .[testing]``), you can check your code by running ``make lint``.
+
+Principles
+~~~~~~~~~~
+
+* Write `valid HTML <https://validator.w3.org/nu/>`_. We target the HTML5 doctype.
+* Attach JavaScript behavior with ``data-`` attributes, rather than classes or IDs.
+* For comments, use Django templates syntax instead of HTML.
```
Comment (jonnyscholes) [inline docs/contributing/html_guidelines.rst]: Could this be a good place to mention the use of semantic html? Something as simple as "Use semantically appropriate HTML elements where possible." Perhaps with a link to http://html5doctor.com/element-index/

File: docs/contributing/developing.rst [hunk 3]
```diff
@@ -179,6 +185,28 @@ IE             Desktop        10 and below
...
+=============  ====================
+
+We aim for Wagtail to work in those environments. Our development standards ensure that the site is usable with other assistive technologies. In practice, testing with assistive technology can be a daunting task that requires specialised training – here are tools we rely on to help identify accessibility issues, to use during development and code reviews:
+
+* `react-axe <https://github.com/dequelabs/react-axe>`_ integrated directly in our build tools, to identify actionable issues.
```
Comment (jonnyscholes) [inline docs/contributing/developing.rst]: Might be worth adding that messages appear in the developer console - I expected to see some kind of UI at first before switching to the console tab.

File: docs/contributing/developing.rst (hunk 1 above)
Comment (thibaudcolas) [reply docs/contributing/developing.rst]: _(for reference, these choices are based on the UK’s GDS [2016 assistive technology survey](https://accessibility.blog.gov.uk/2016/11/01/results-of-the-2016-gov-uk-assistive-technology-survey/), with further guidance on what to test with if you want to do [testing at no cost](https://accessibility.blog.gov.uk/2018/09/27/assistive-technology-tools-you-can-use-at-no-cost/))_

I'm in two minds about it. It feels really relevant for mobile screenreader(s) / accessibility tools to be on there, and I regularly use my phone's VoiceOver when doing testing. At the same time, Wagtail’s mobile experience is quite poor for sighted users, so aiming for screen reader support there feels like a step too far. I don't know how to decide :/

File: docs/contributing/html_guidelines.rst (hunk 2 above)
Comment (thibaudcolas) [reply docs/contributing/html_guidelines.rst]: Yes!

**Comment by thibaudcolas** [comment]
Thanks for the super fast review @jonnyscholes 😊 I'll leave this open a bit longer, make the changes you suggested, and most likely merge this early next week unless there is more feedback / discussion.

**Comment by thibaudcolas** [comment]
I've added feedback from @jonnyscholes, will merge this in a few hours unless there is more feedback.

**Pull request merged**